pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -248,17 +248,39 @@ func (c *Reconciler) createConfiguration(service *v1alpha1.Service) (*v1alpha1.C
 	return c.ServingClientSet.ServingV1alpha1().Configurations(service.Namespace).Create(cfg)
 }
 
+func configSemanticEquals(config, desiredConfig *v1alpha1.Configuration) bool {
+	return equality.Semantic.DeepEqual(desiredConfig.Spec, config.Spec) &&
+		equality.Semantic.DeepEqual(desiredConfig.ObjectMeta.Labels, config.ObjectMeta.Labels)
+}
+
+// mergeRouteLabel sets desiredConfig[serving.RouteLabelKey] to same
+// as config[serving.RouteLabelKey], so that we do nothing about
+// configuration label serving.RouteLabelKey in our reconciliation.
+func mergeRouteLabel(desiredConfig, config *v1alpha1.Configuration) {
+	routeLabel, existed := config.ObjectMeta.Labels[serving.RouteLabelKey]
+	if !existed {
+		delete(desiredConfig.ObjectMeta.Labels, serving.RouteLabelKey)
+	} else {
+		desiredConfig.ObjectMeta.Labels[serving.RouteLabelKey] = routeLabel
+	}
+}
+
 func (c *Reconciler) reconcileConfiguration(ctx context.Context, service *v1alpha1.Service, config *v1alpha1.Configuration) (*v1alpha1.Configuration, error) {
 	logger := logging.FromContext(ctx)
 	desiredConfig, err := resources.MakeConfiguration(service)
 	if err != nil {
 		return nil, err
 	}
+	// Route label is automatically set by another reconciler.  We
+	// want to ignore that label in our reconciliation here by setting
+	// desiredConfig[serving.RouteLabelKey] to the same as
+	// config[serving.RouteLabelKey].
+	mergeRouteLabel(desiredConfig, config)
```

Comment:
This took me a while to understand. This does not seem to be doing a merge and seems very specific to the RouteLabelKey. If I understand correctly it essentially ignores anything under this key by setting it to the same as the existing config. A function like below would have been easier for me to grok:

// Does not reconcile labels under any of the keys passed in. Existing config remains unmodified for these labels. Currently used for route label as it is automatically set by another reconciler.
ignoreLabelKeys(desiredConfig, config, LabelKeys...)

Reply:
changed to 

ignoreRouteLabelChange(...)

I didn't go with the more generic version since we only have one label to ignore.